.github/workflows/ruby.yml

```diff
@@ -1,32 +1,70 @@
 name: Test Suite
 
-on: push
+# Run against all commits and pull requests.
+on: [ push, pull_request ]
 
 jobs:
-  test:
+  test_matrix:
 
-    runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        ruby-version: ['3.0', 2.7]
+        os:
+          - ubuntu
+          - macos
+        ruby:
+          - 2.6
+          - 2.7
+          - 3.0
```

Comment:
Previously `3.0` was a string due to bugs in the Github Action code, but it seems to work fine without it currently. I'm guessing whatever the issue was got solved upstream since then.

Reply:
This is correct. I squashed all my test commits down when I worked on this, but my early efforts used a number and it would crash on me.